avro/avro.bzl

```diff
@@ -268,11 +268,14 @@ def _gen_impl(ctx):
         "find {gen_dir} -exec touch -t 198001010000 {{}} \\;".format(
           gen_dir=gen_dir
         ),
-        "{jar} cMf {output} -C {gen_dir} .".format(
-          jar="%s/bin/jar" % ctx.attr._jdk[java_common.JavaRuntimeInfo].java_home,
-          output=ctx.outputs.codegen.path,
-          gen_dir=gen_dir
-        )
+        "base_dir=$(pwd)",
+        "pushd {gen_dir}".format(gen_dir = gen_dir),
+        # Sort the entries when zipping in order to guarantee deterministic outputs.
+        # Note that we use zip instead of jar because jar does not seem to respect insert ordering.
+        "find * | sort |  xargs \"${{base_dir}}/{zipper}\" c \"${{base_dir}}/{output}\"".format(
```

Comment:
I think this'd be better as `find .` instead of `find *`. No need for the expansion, and `find .` will discover hidden files as well.

Reply:
yeah, I was trying to avoid the leading `./` in the output but `find . -printf` is a cleaner way